backend/tournesol/tests/test_api_contributor_recommendations.py

```diff
@@ -0,0 +1,335 @@
+        )
+        # the private endpoint must return one recommendation
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data["count"], 1)
+
+        response = self.client.get(
+            f"/users/{self.user1.username}/recommendations/{new_poll.name}",
+            format="json",
+        )
+        # the public endpoint must return one recommendation
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data["count"], 1)
+
+    def test_users_cant_list_with_non_existing_user(self):
+        """
+        Anonymous and authenticated users can't list recommendations of a
+        non-existing user.
+        """
+        response = self.client.get(
+            f"/users/non-existing/recommendations/{self.poll.name}", format="json"
+        )
+        self.assertEqual(response.status_code, 404)
+
+        self.client.force_authenticate(self.user1)
+        response = self.client.get(
+            f"/users/non-existing/recommendations/{self.poll.name}", format="json"
+        )
+        self.assertEqual(response.status_code, 404)
+
+    def test_users_cant_list_with_non_existing_poll(self):
+        """
+        Anonymous and authenticated users can't list recommendations of a
+        non-existing poll.
+        """
+        response = self.client.get(
+            f"/users/{self.user1.username}/recommendations/non-existing", format="json"
+        )
+        self.assertEqual(response.status_code, 404)
+
+        self.client.force_authenticate(self.user1)
+        response = self.client.get(
+            f"/users/{self.user1.username}/recommendations/non-existing", format="json"
+        )
+        self.assertEqual(response.status_code, 404)
+
+    def test_users_can_list_with_param_unsafe(self):
+        """
+        Anonymous and authenticated users can filter recommendations with the
+        `unsafe` URL parameter.
+        """
+        user = UserFactory()
+        entity = EntityFactory()
+        rating = ContributorRatingFactory(user=user, entity=entity, is_public=True)
+        ContributorRatingCriteriaScoreFactory(
+            contributor_rating=rating,
+            criteria=self.criterion,
+            score=-1,
+        )
+
+        # anonymous checks
+        response = self.client.get(
+            f"/users/{user.username}/recommendations/{self.poll.name}", format="json"
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data["count"], 0)
+
+        response = self.client.get(
+            f"/users/{user.username}/recommendations/{self.poll.name}?unsafe=true",
+            format="json",
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data["count"], 1)
+
+        # authenticated checks
+        self.client.force_authenticate(self.user1)
+        response = self.client.get(
+            f"/users/{user.username}/recommendations/{self.poll.name}", format="json"
```

Comment:
here too, is it rather "users/me/recommendations/{self.poll.name}" ?

Reply:
Here too, I'm performing checks with both anonymous and authenticated users, that's making I'm making several requests to the same endpoint.